### ISSUE TITLE
significant degradation on mesh for subsequent frames after the first initial optimization

### ISSUE THREAD

**Issue by Ruxie189** [XuanchenLi/Topo4D#10]
Hello, 

Thanks for the code. I was trying to optimize topo4D for an inhouse dataset but I am observing significant degradation on mesh for subsequent frames after the first initial optimization of 7000 iterations. The images after perfectly aligned to the camera parameters (sanity checked for 0th iteration). I also use the facial partitions provided here because i converted the dataset to topo4d's topology. 

I have attached the mesh for your reference. Can you please help me with this?

![Image](https://github.com/user-attachments/assets/bfc16aef-d958-4120-bbc1-1e6baa71c071)

![Image](https://github.com/user-attachments/assets/d38b2209-e2f0-4f3b-bd5b-d94bba94920a)

**Comment by Ruxie189** [comment]
For reference the frame below is after 7000 iterations of initial optimization and the one above is for next frame

**Comment by XuanchenLi** [comment]
This should be due to the mesh scale being too small. Such degradation occurs if the values of the vertices are too small and the learning rate is too high. You can adjust the scale of your mesh to a size similar to the example we provided and then modify your cameras accordingly.

**Comment by Ruxie189** [comment]
Hii, thank you for your response! Scaling up worked. Could you also let me know what is target_colors_1K = [torch.tile(torch.tensor(cmap[label_index]).reshape(3, 1, 1), (1, 1024, 750)).cuda() for label_index in range(14)] in line 808, helpers.py. Is it the target colour labels for sample dataset for the data provided because I don't think its executing?

**Comment by XuanchenLi** [comment]
This line is used to pre allocate the label tensor used in "get_mask" (line 810) for the 1K resolution input image. Such code is indeed not user-friendly, and I forgot to rewrite it. I have modified this part, thank you for your question.

**Comment by Ruxie189** [comment]
Thank you so much for the clarification! I am closing the issue as my request is resolved.
